### ISSUE TITLE
Preserve fallback behaviour for non-project files if their parent git repository not the one we wanted

### ISSUE THREAD

**Issue by kutsan** [airblade/vim-rooter#69]
My home directory is a git repository for my dotfiles. So, using this plugin for non-project files will always fallback to home, because their root git repository is home.

For example if my current working directory is `~/Documents`, editing files under `~/Documents` will always set CWD to home which isn't what I want. A option to tell the plugin ignore the home git repository would be nice or ignore home completely by default if home is a git repository. A lot of people using it that way.

**Comment by airblade** [comment]
In your situation I would just configure vim-rooter to recognise your `~/Documents` directory as a "root" directory:

```viml
let g:rooter_patterns = ['Documents/', '.git', '.git/', '_darcs/', '.hg/', '.bzr/', '.svn/']
```

**Comment by kutsan** [comment]
Well, I don't think it's a solution to be honest. Am I supposed to add every directory as root pattern? It's so annoying because I don't always edit non-project files under the same directory. `~/Documents` was only an example. I would rather choose a dynamic implementation.

Even what will happen, if I edit a non-project file like `~/Documents/somedirectory/somefile.vim`? Root will be still `~/Documents` which is still problematic.

Please, re-open the issue.

**Comment by airblade** [comment]
A dynamic implementation could be to add an empty file, say `.root`, in directories which you don't want the plugin to search above, and have `let g:rooter_patterns = ['.root', ...]`.

I suppose one could add the ability to ignore specified directories but it would not be top of my priority list.

Alternatively you could maybe configure `g:rooter_targets` not to include your non-project file extensions.

**Comment by kutsan** [comment]
It's still the same problem. I still have to still `.root` file every single directory. It's just workaround and things get mess over time.

I understand, it's not priority at this moment but re-opening the issue could help potential contributors to find this problem. Maybe they can help.

**Comment by airblade** [comment]
Nearly seven years later this is addressed (I hope!) by 9c6ce62.
